rest/route/annotations.go

```diff
@@ -116,3 +117,101 @@ func getAPIAnnotationsForTaskIds(taskIds []string, allExecutions bool) gimlet.Re
 
 	return gimlet.NewJSONResponse(res)
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// GET /rest/v2/task/{task_id}/{task_execution}/annotation
+
+type annotationByTaskHandler struct {
+	taskId             string
+	fetchAllExecutions bool
+	execution          int
+	sc                 data.Connector
+}
+
+func makeFetchAnnotationByTask(sc data.Connector) gimlet.RouteHandler {
+	return &annotationByTaskHandler{
+		sc: sc,
+	}
+}
+
+func (h *annotationByTaskHandler) Factory() gimlet.RouteHandler {
+	return &annotationByTaskHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *annotationByTaskHandler) Parse(ctx context.Context, r *http.Request) error {
+	var err error
+
+	h.taskId = gimlet.GetVars(r)["task_id"]
+	if h.taskId == "" {
+		return gimlet.ErrorResponse{
+			Message:    "task ID cannot be empty",
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+
+	vals := r.URL.Query()
+	h.fetchAllExecutions = vals.Get("fetch_all_executions") == "true"
+	execution := vals.Get("execution")
+
+	if execution != "" && h.fetchAllExecutions == true {
+		return gimlet.ErrorResponse{
+			Message:    "fetchAllExecutions=true cannot be combined with execution={task_execution}",
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+
+	if execution != "" {
+		h.execution, err = strconv.Atoi(execution)
+		if err != nil {
+			return gimlet.ErrorResponse{
+				Message:    "Invalid execution",
+				StatusCode: http.StatusBadRequest,
+			}
+		}
+	} else {
+		// since an int in go defaults to 0, we won't know if the user
+		// specifically wanted execution 0, or if they want the latest.
+		// we use -1 to indicate "not specified"
+		h.execution = -1
+	}
+
+	//todo: don't allow giving both fetchall and an execution
+
+	return nil
+}
+
+func (h *annotationByTaskHandler) Run(ctx context.Context) gimlet.Responder {
+	// get a specific execution
+	if !h.fetchAllExecutions && h.execution != -1 {
+		a, err := annotations.FindAnnotationByTaskIdAndExecution(h.taskId, h.execution)
+		if err != nil {
+			return gimlet.NewJSONInternalErrorResponse(errors.Wrap(err, "error finding task annotation"))
+		}
+		if a == nil {
+			return gimlet.NewJSONResponse(model.APITaskAnnotation{})
+		}
+		taskAnnotation := model.APITaskAnnotationBuildFromService(*a)
+		return gimlet.NewJSONResponse([]model.APITaskAnnotation{*taskAnnotation})
+	}
+
+	allAnnotations, err := annotations.FindAnnotationsByTaskId(h.taskId)
+	if err != nil {
+		return gimlet.NewJSONInternalErrorResponse(errors.Wrap(err, "error finding task annotations"))
+	}
+	// get the latest execution
+	annotationsToReturn := allAnnotations
+	if !h.fetchAllExecutions && h.execution == -1 {
```

Comment:
because you already checked `!fetchAllExecutions && h.execution != -1` at line 188, we know anything that is `!fetchAllExecutions` is also `h.execution == -1`, so you can drop that part of the condition

Reply:
I know, but I left it in there because I feel like this makes it more readable. It also adds an extra check in case the code morphs over time. 
Lmk if you feel strongly about this. If you do, I'll remove it.

Reply:
I think it's kind of confusing because it implies more cases than there are. Even the check at 186 isn't as intuitive since fetchAllExecutions can't be true if execution > -1. Really the cases are mutually exclusive and it's definitely fine to assume that given Parse.

Reply:
that makes sense. I'll fix it.